bsms.mediawiki

```diff
@@ -0,0 +1,153 @@
+
+====Round 2====
+
+=====Coordinator=====
+
+* The Coordinator gathers key records from all participating Signers. Abort the setup if the wallet setup session has expired.
+* For each key record, the Coordinator decrypts it using <tt>ENCRYPTION_KEY</tt>. The Coordinator verifies that the included <tt>SIG</tt> is valid given the <tt>KEY</tt>.
+* If all key records look good, the Coordinator fills in all necessary information to generate a descriptor record, which is simply the descriptor string plus a <tt>CHECKSUM</tt>, all in one line. The <tt>CHECKSUM</tt> has [https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md#checksums BECH32 encoding].
+* The Coordinator encrypts this descriptor record with <tt>ENCRYPTION_KEY</tt>.
+* The Coordinator sends the encrypted descriptor record to all participating Signers.
+
+=====Signer=====
+
+* The Signer imports the descriptor record, decrypts it using the <tt>ENCRYPTION_KEY</tt> derived from the open session.
+* The Signer calculates and verifies the descriptor’s <tt>CHECKSUM</tt>. Abort the setup if the <tt>CHECKSUM</tt> is incorrect.
+* The Signer checks whether one of the <tt>KEY</tt>s in the descriptor belongs to it, using path and fingerprint information included in the descriptor. The check must perform an exact match on the <tt>KEY</tt>s, and not using shortcuts such as matching fingerprints (which is trivial to spoof). Abort the setup if it doesn’t detect its own <tt>KEY</tt>.
+* For confirmation, the Signer must display to the user the <tt>CHECKSUM</tt>, plus other configurations, such as <tt>M</tt> and <tt>N</tt>. The total number of Signers, <tt>N</tt>, is important to prevent a <tt>KEY</tt> insertion attack. All participating Signers should be able to display the same confirmation.
+* If all checks pass, the Signer persists the descriptor record in its storage.
+* The Signer can choose to further restrict post-XPUB derivation paths, such as to those defined in [https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki BIP-0044].
+* The Signer should subsequently use the descriptor to generate and verify receive and change addresses.
+
+This completes the setup.
+
+===Encryption===
+
+====The Token====
+We define three modes of encryption.
+
+# <tt>NO_ENCRYPTION</tt> : the <tt>TOKEN</tt> is set to <tt>0</tt>. Encryption is disabled.
+# <tt>STANDARD</tt> : the <tt>TOKEN</tt> is a 64-bit nonce.
+# <tt>EXTENDED</tt> : the <tt>TOKEN</tt> is a 128-bit nonce.
+
+The <tt>TOKEN</tt> can be converted to one of these formats:
+* A mnemonic phrase using [https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki BIP-0039] word list (6 words in <tt>STANDARD</tt> mode, 12 words in <tt>EXTENDED</tt> mode)
```

Comment:
I am against using the mnemonic form to encode the 128-bit nonce. This can lead to confusion where people might use this 12 words to store their funds as this will be indistinguishable from the standard BIP39 recovery seed.

Reply:
that's a good point. what do you think about 9 words instead? cc @dgpv

Reply:
@hugohn I've commented on this above, but I forgot to click 'sumbit review' button and my comments were in the 'pending' state, invisible to others until now :-/

Reply:
9 words are fine

Reply:
cool, I'll change it to 9 words.

Reply:
What about encoding the token in bech32 using a unique prefix? IMHO the redundancy and human readability of the mnemonic form just isn't a good fit here, I expect the common case is to copy and paste this from an external safe channel.

Reply:
@juestr - the idea here is that you should be able to enter the code (numbers of words) directly on an airgapped device

Reply:
Good point, a tagged bech32 value might still improve UX for QR codes etc.